src/util/duration.h

```diff
@@ -95,11 +95,11 @@ class DurationBase {
     static QChar kDecimalSeparator;
 
   protected:
-    explicit constexpr DurationBase(qint64 durationNanos)
-        : m_durationNanos(durationNanos) {
+    explicit constexpr DurationBase(double durationNanos)
+            : m_durationNanos(durationNanos) {
```

Comment:
Please add an assertion to check that `durationNanos >= 0`.

Reply:
That would change the class semantically. Negative durations are explicitly checked in our tests. I agree that negative durations seems counterintuitive, but such a change should be a different PR IMHO.